breadbox/alembic/versions/e593fefbe9fc_move_column_refences.py

```diff
@@ -0,0 +1,126 @@
+    )
+    # Do the actual updates
+    op.execute(
+        "update dimension set references_dimension_type_name = (select references_dimension_type_name from tmp_references_to_copy where id = dimension_id) where exists (select references_dimension_type_name from tmp_references_to_copy where id = dimension_id) "
+    )
+
+    # now we can clean up and drop tables
+    op.drop_table("tmp_references_to_copy")
+    op.drop_table("dataset_reference")
+
+    # save original property_to_index table
+    op.rename_table("property_to_index", "old_property_to_index")
+    # create new version with dimension_type_name instead of dataset_id
+    op.create_table(
+        "property_to_index",
+        sa.Column("id", sa.String(length=36), nullable=False),
+        sa.Column("dimension_type_name", sa.String(), nullable=False),
+        sa.Column("property", sa.String(), nullable=False),
+        sa.Column("group_id", sa.String(), nullable=False),
+        sa.ForeignKeyConstraint(
+            ["dimension_type_name"],
+            ["dimension_type.name"],
+            name=op.f("fk_property_to_index2_dimension_type_name_dimension_type"),
+            ondelete="CASCADE",
+        ),
+        sa.ForeignKeyConstraint(
+            ["group_id"],
+            ["group.id"],
+            name=op.f("fk_property_to_index2_group_id_group"),
+        ),
+        sa.PrimaryKeyConstraint("id", name=op.f("pk_property_to_index2")),
+    )
+    # copy rows from old table into new table
+    op.execute(
+        "insert into property_to_index ('id', 'dimension_type_name', 'property', 'group_id') select pti.id, dt.name, pti.property, pti.group_id from old_property_to_index pti join dataset d on pti.dataset_id = d.id join dimension_type dt on dt.dataset_id = d.id"
+    )
+    # clean up the old version
+    op.drop_table("old_property_to_index")
+
+    # figure out how to update property_to_index
+    # op.execute()
+    #
+    #
+    # with op.batch_alter_table('property_to_index', schema=None) as batch_op:
+    #     batch_op.add_column(sa.Column('dimension_type_name', sa.String(), nullable=False))
+    #     batch_op.drop_constraint('fk_property_to_index_dataset_id_dataset', type_='foreignkey')
+    #     batch_op.create_foreign_key(batch_op.f('fk_property_to_index_dimension_type_name_dimension_type'), 'dimension_type', ['dimension_type_name'], ['name'], ondelete='CASCADE')
+    #     batch_op.drop_column('dataset_id')
+    #
+
+    # drop search index and recreate it. We should add a step to deploy to populate this if it's empty
```

Comment:
I see your note here: 
>  drop search index and recreate it. We should add a step to deploy to populate this if it's empty

And I also see that you removed the command `populate-index` from `commands.py`. Is there another way to populate this search index during a deployment?

Reply:
Ah, right. I forgot to circle back and do that part. 

My plan was to add a step that's always run as part of migrations. The new step would check to see if the index is empty, and if so, re-run it on all dimension_types. If the table _isn't_ empty, then it'd do nothing. 

(It would allow us to "invalidate" the index by deleting the contents. I like this for the simplicity. I don't like this for how implicit it is... For example, I don't think it'll be obvious to anyone that this behavior exists. Anyway, I'll give it a try because I don't have better ideas.)